### PR TITLE
pacemaker code refactoring

### PR DESCRIPTION
**Pull request by Zipfer**
Please attentively look at this refactored code  for mistakes

### CODE REVIEW

**Comment by bogdando** [comment]
-1
Please make sure all tests which using dump_cib are updated

File: deployment/puppet/corosync/lib/puppet/provider/service/pacemaker.rb [hunk 1]
```diff
@@ -118,15 +114,17 @@ def self.get_node_state_stonith(ha_state,in_ccm,crmd,join,expected,shutdown)
...
     elsif crmd == 'online'
+
       if join == 'member'
         state = :online
       else
         state = :offline
       end
-    elsif !(shutdown == "0")
+
+    elsif !(shutdown == '0')
```
Comment (warpc) [inline deployment/puppet/corosync/lib/puppet/provider/service/pacemaker.rb]:
``` ryby
elsif shutdown != '0'
```

?

File: deployment/puppet/corosync/lib/puppet/provider/service/pacemaker.rb [hunk 2]
```diff
@@ -173,6 +169,7 @@ def self.get_nodes
   def get_last_successful_operations
     self.class.get_cib
     self.class.get_nodes
+    return @last_successful_operation unless @last_successful_operation.empty?
     @last_successful_operations = []
```
Comment (warpc) [inline deployment/puppet/corosync/lib/puppet/provider/service/pacemaker.rb]:
``` ruby
@last_successful_operations ||= []
return @last_successful_operation unless @last_successful_operation.empty?
```

If stay as now we get error.

``` ruby
undefined method `empty?' for nil:NilClass
```

It is my fall, sorry)

File: deployment/puppet/corosync/lib/puppet/provider/service/pacemaker.rb [hunk 3]
```diff
@@ -270,14 +264,12 @@ def restart
...
-    debug("getting last operations")
-    get_last_successful_operations
-    if @last_successful_operations.any? {|op| ['start','promote'].include?(op)}
+  def status
+    crm_resource('--resource', get_service_name, '--cleanup', '--node', `uname -n`.chomp)
+    debug('getting last operations')
+    if get_last_successful_operations.any? {|op| ['start','promote'].include?(op)}
       return :running
-    elsif @last_successful_operations.all? {|op| op == 'stop'} or @last_successful_operations.empty?
+    elsif get_last_successful_operations.all? {|op| op == 'stop'} or get_last_successful_operations.empty?
```
Comment (warpc) [inline deployment/puppet/corosync/lib/puppet/provider/service/pacemaker.rb]: ||

File: deployment/puppet/corosync/lib/puppet/provider/service/pacemaker.rb [hunk 4]
```diff
@@ -203,9 +199,7 @@ def get_last_successful_operations
         debug("last operations: #{last_op.attributes['operation']}")
         last_successful_op = last_op.attributes['operation']
       else
-        if last_op.attributes['rc-code'].to_i == 0
-          last_successful_op = 'start'
-        elsif  last_op.attributes['rc-code'].to_i == 8
+        if (last_op.attributes['rc-code'].to_i == 0) || (last_op.attributes['rc-code'].to_i == 8)
```
Comment (warpc) [inline deployment/puppet/corosync/lib/puppet/provider/service/pacemaker.rb]:
``` ruby
if [0,8].include? last_op.attributes['rc-code'].to_i
```

File: deployment/puppet/corosync/lib/puppet/provider/service/pacemaker.rb (hunk 1 above)
Comment (Zipfer) [inline deployment/puppet/corosync/lib/puppet/provider/service/pacemaker.rb]: of course yes)
I change this for other code

File: deployment/puppet/corosync/lib/puppet/provider/service/pacemaker.rb [hunk 5]
```diff
@@ -106,7 +102,7 @@ def self.get_node_state_stonith(ha_state,in_ccm,crmd,join,expected,shutdown)
       else
         state = :unclean
       end
-    elsif !(in_ccm == "true") && (ha_state =='dead') && (crmd == 'offline') && !(shutdown == 0)
+    elsif !(in_ccm == 'true') && (ha_state =='dead') && (crmd == 'offline') && !(shutdown == 0)
```
Comment (warpc) [inline deployment/puppet/corosync/lib/puppet/provider/service/pacemaker.rb]:
``` ruby
elsif in_ccm != 'true' && ha_state =='dead' && crmd == 'offline' && shutdown != 0
```

**Comment by warpc** [comment]
+1

**Comment by xenolog** [comment]
retest this please

**Pull request closed without merging**